src/scanspec/specs.py

```diff
@@ -619,17 +596,18 @@ def calculate(self, bounds=True, nested=False) -> List[Frames[Axis]]:
             self._spiral_from_indexes, self.axes(), self.num, bounds
         )
 
-    @alternative_constructor
```

Comment:
Proxy thread for discussion in previous PR: https://github.com/dls-controls/scanspec/pull/54#discussion_r1007061793

Reply:
I'm now tempted to say that we don't need alternative constructors for the REST interface, since REST is all about specifying canonical data models. In other words, the model of a `Line` is `axis`, `start`, `stop` and `num`, and if DAQ want to specify it a different way then the conversion is our responsibility client-side. This potentially creates a little duplication but also ensures separation of concerns. Happy to discuss though. If we do loose alternative constructors, we may also wish to ask whether the `/valid` endpoint is even needed any more.

Reply:
From discussion: Alternative constructors may be needed, will take the agile approach and reintroduce them when we make API clients. The behaviour is still available in native Python via `classmethod`s.